src/ast/semantic_analyser.cpp

```diff
@@ -1285,11 +1309,19 @@ bool SemanticAnalyser::check_assignment(const Call &call, bool want_map, bool wa
       return false;
     }
   }
+  else if (want_map_key)
+  {
+    if (!call.key_for_map)
+    {
+      err_ << call.func << "() should be assigned to a variable" << std::endl;
```

Comment:
s/variable/map key/  right?

Reply:
That might read weird. Maybe
```suggestion
      err_ << call.func << "() should be used as a map key" << std::endl;
```